banzai_floyds/extract.py

```diff
@@ -0,0 +1,190 @@
+from banzai.stages import Stage
+import numpy as np
+from astropy.table import Table
+from banzai_floyds.matched_filter import maximize_match_filter
+from numpy.polynomial.legendre import Legendre
+from banzai_floyds.utils.fitting_utils import gauss, fwhm_to_sigma
+
+
+def profile_gauss_fixed_width(params, x, sigma):
+    center, background_level = params
+    return gauss(x, center, sigma) + background_level
+
+
+def background_fixed_profile_center(params, x, center):
+    sigma, *coeffs = params
+    background = Legendre(coef=coeffs, domain=(np.min(x), np.max(x)))
+    return gauss(x, center, sigma) + background(x)
+
+
+def fit_profile(data, uncertainty, wavelengths, orders, wavelength_bins, profile_width=4):
+    x2d, y2d = np.arange(data.shape[1]), np.arange(data.shape[0])
+
+    trace_centers = []
+    # for each order
+    order_values = np.unique(orders.data)
+    order_values = order_values[order_values != 0]
+    for order_value, order_wavlengths in zip(order_values, wavelength_bins):
+        in_order = orders.data == order_value
+        # y = (y2d - orders.center(x2d))[in_order]
+        y = orders.center(x2d)[order_value]
```

Comment:
I'm not sure exactly what we are trying to do here. 
Is this supposed to be an array with the dimensions of data?